test/unit/cuda/util/cuda_gtest_main.cpp

```diff
@@ -0,0 +1,9 @@
+#include <gtest/gtest.h>
+#include <micm/cuda/util/cuda_util.cuh>
+
+int main(int argc, char **argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+    int result = RUN_ALL_TESTS();
+    micm::cuda::CudaStreamSingleton::GetInstance().CleanUp(); // Ensure cleanup
```

Comment:
Is this necessary? I think if you define the destructor as I suggested above that we would no longer need this file because the destructor would always run and cleanup the streams

Reply:
See comment above.